engine_architecture.tex

```diff
@@ -0,0 +1,75 @@
+
+\subsection{Components of a solution}
+% Opisanie elementów z jakich się składa rozgrywka m.in.: kafelki i ich elementy, gracze, a także uwzględnione rozszerzenia gry Carcassonne.
+
+The game Carcassone is played by placing tiles \cite{CarcassoneRules}. Each tile is represented by a set of features on it. All features are defined by their type, such as road, city, field or monastery, a modifier that modifies the calculation process, and the sides which indicate on which edges the feature is present. In this project there is only one modifier used - the shield. The engine was designed to accommodate game extensions. 
+
+A feature's side is a byte sized value, in which each bit defines whether the feature is present on a specific half of the tile's edge as shown on Figure \ref{fig:SIDES}. Features such as roads, are represented by a whole edge formed by combining two sides e.g., the TopLeft and TopRight sides.
+\begin{figure}
+    \centering
+    \begin{tikzpicture}
+        %top line
+        \draw [blue] (-3,3) -- (0,3);
+        \draw [red] (0,3) -- (3,3);
+        %right line
+        \draw [blue] (3,3) -- (3,0);
+        \draw [red] (3,0) -- (3,-3);
+        %bottom line
+        \draw [red] (3,-3) -- (0,-3);
+        \draw [blue] (0,-3) -- (-3,-3);
+        %left line
+        \draw [blue] (-3,-3) -- (-3,0);
+        \draw [red] (-3,0) -- (-3,3);
+        
+        \node at (-1.5, 2.7) {TopLeft};
+        \node at (1.5, 2.7) {TopRight};
+        
+        \node [rotate=90] at (2.8, 1.5) {RightTop};
+        \node [rotate=90] at (2.8, -1.5) {RightBottom};
+        
+        \node at (-1.5, -2.8) {BottomLeft};
+        \node at (1.5, -2.8) {BottomRight};
+        
+	% czy może obrót 90stopni?
+        \node [rotate=270] at (-2.8, 1.5) {LeftTop};
+        \node [rotate=270] at (-2.8, -1.5) {LeftBottom};
+    \end{tikzpicture}
+    \caption{Side representation} \label{fig:SIDES}
+\end{figure}
+
+The tiles described do not contain information about their position or meeple placement. This information is located in the PlacedTile struct. The board is represented as a slice of placed tiles ordered by their placement, moreover it contains the used tile set, for verification whether the tile a player wants to place is legal. Additionally, board includes a city manager used for merging cities, scoring and checking for legal meeples placement in the cities.
+
+A single game is represented by a board, shuffled deck of tiles, and players taking their turns. Each player knows their score, and the number of meeples of each type available to them.
```

Comment:
> [...] and players taking their turns. [...]

I think "and players partaking in it" or "and a ?slice? of players" would be more fitting. "Taking their turns" is an action so it is not included in the game representation per se.

Reply:
changed for "and players partaking in it"